rust/src/dns/log.rs

```diff
@@ -584,92 +550,132 @@ fn dns_log_json_answer(
+    };
+
+    // The internal Suricata transaction ID.
+    jb.set_uint("tx_id", tx.id - 1)?;
+
+    // The on the wire DNS transaction ID.
+    jb.set_uint("id", tx.tx_id() as u64)?;
+
+    // Log header fields. Should this be a sub-object?
+    let header = &message.header;
+    jb.set_string("flags", format!("{:x}", header.flags).as_str())?;
+    if header.flags & 0x8000 != 0 {
+        jb.set_bool("qr", true)?;
+    }
+    if header.flags & 0x0400 != 0 {
+        jb.set_bool("aa", true)?;
+    }
+    if header.flags & 0x0200 != 0 {
+        jb.set_bool("tc", true)?;
+    }
+    if header.flags & 0x0100 != 0 {
+        jb.set_bool("rd", true)?;
+    }
+    if header.flags & 0x0080 != 0 {
+        jb.set_bool("ra", true)?;
+    }
+    if header.flags & 0x0040 != 0 {
+        jb.set_bool("z", true)?;
+    }
+    let opcode = ((header.flags >> 11) & 0xf) as u8;
+    jb.set_uint("opcode", opcode as u64)?;
+    jb.set_string("rcode", &dns_rcode_string(header.flags))?;
+
+    if !message.queries.is_empty() {
+        jb.open_array("queries")?;
+        for query in &message.queries {
             if dns_log_rrtype_enabled(query.rrtype, flags) {
-                jb.set_string("type", "query")?;
-                jb.set_uint("id", request.header.tx_id as u64)?;
-                jb.set_string_from_bytes("rrname", &query.name)?;
-                jb.set_string("rrtype", &dns_rrtype_string(query.rrtype))?;
-                jb.set_uint("tx_id", tx.id - 1)?;
-                if request.header.flags & 0x0040 != 0 {
-                    jb.set_bool("z", true)?;
-                }
-                let opcode = ((request.header.flags >> 11) & 0xf) as u8;
-                jb.set_uint("opcode", opcode as u64)?;
-                return Ok(true);
+                jb.start_object()?
+                    .set_string_from_bytes("rrname", &query.name)?
+                    .set_string("rrtype", &dns_rrtype_string(query.rrtype))?
+                    .close()?;
             }
         }
+        jb.close()?;
     }
 
-    return Ok(false);
-}
+    // For now, stop logging if a request to limit changes from 7.0
+    // behavior. But revisit as it could be useful to log this data
+    // when its part of a request.
+    if tx.is_request() {
+        return Ok(());
+    }
 
-#[no_mangle]
-pub extern "C" fn SCDnsLogJsonQuery(
-    tx: &DNSTransaction, i: u16, flags: u64, jb: &mut JsonBuilder,
-) -> bool {
-    match dns_log_query(tx, i, flags, jb) {
-        Ok(false) | Err(_) => {
-            return false;
-        }
-        Ok(true) => {
-            return true;
+    if !message.answers.is_empty() {
+        dns_log_json_answers(jb, message, flags)?;
+    }
+
+    if !message.authorities.is_empty() {
+        jb.open_array("authorities")?;
+        for auth in &message.authorities {
+            let auth_detail = dns_log_json_answer_detail(auth)?;
+            jb.append_object(&auth_detail)?;
         }
+        jb.close()?;
     }
+
+    // TODO:
+    // - additionals
```

Comment:
Is there a ticket for this ?

Reply:
Yes, and a PR: https://github.com/OISF/suricata/pull/11263

Reply:
New PR has this work merged in.